internal/server/GameLoop.go

```diff
@@ -17,59 +14,23 @@ func (s *Server) RunGameLoop() {
 }
 
 func (s *Server) LootboxCheckAndDistributions() {
-	EPSILON := 0.1          // Collision detection threshold
-	COLOURMULTIPLIER := 1.2 // Multiplier for colour match
-	// a > c > b
-	a := 1.0
-	b := 0.5
-	c := 0.3
-	/**
-	Agents to focus on just meeting needs without hoarding: a to be high relative to b and c.
-	Discourage agents from appropriating less than their needs: c high.
-	Encourage agents to gather as many resources as possible: b high.
-	**/
 	for bikeid, megabike := range s.GetMegaBikes() {
 		for lootid, lootbox := range s.GetLootBoxes() {
-			if utils.CheckCollision(megabike.GetPosition(), lootbox.GetPosition(), EPSILON) {
+			if megabike.CheckForCollision(lootbox) {
 				// Collision detected
 				fmt.Printf("Collision detected between MegaBike %s and LootBox %s \n", bikeid, lootid)
 				agents := megabike.GetAgents()
-				totalUtility := 0.0
-				utilityMap := make(map[uuid.UUID]float64)
-				// Get total amount of attributes for each agent on the bike
-				totalG := 0.0
-				totalQ := 0.0
-				totalP := 0.0
-				totalR := 0.0
-				for _, agent := range agents {
-					totalG += agent.GetEnergyLevel()
-					totalQ += agent.GetResourceNeed()
-					totalP += agent.GetResourceProvision()
-					totalR += agent.GetResourceAppropriation()
-				}
-				// Calculate utility for each agent on the bike
-				for _, agent := range agents {
-					g := agent.GetEnergyLevel() / totalG
-					q := agent.GetResourceNeed() / totalQ
-					p := agent.GetResourceProvision() / totalP
-					r := agent.GetResourceAppropriation() / totalR
-					R := r + (g - p) // Accrued resources
+				totAgents := len(agents)
 
-					var u float64 // Utility
-					if R >= q {
-						u = a*q + b*(R-q)
-					} else {
-						u = a*R - c*(q-R)
-					}
-					if agent.GetColour() == lootbox.GetColour() {
-						u *= COLOURMULTIPLIER
-					}
-					utilityMap[agent.GetID()] = u
-					totalUtility += u
-				}
-				// Distribute loot based on utility as a share of the total utility
 				for _, agent := range agents {
-					utilityShare := utilityMap[agent.GetID()] / totalUtility
+					// this function allows the agent to decide on its allocation parameters
+					// these are the parameters that we want to be considered while carrying out
+					// the elected protocol for resource allocation
+					agent.SetAllocationParameters()
+
+					// in the MVP  the allocation parameters are ignored and
+					// the utility share will simply be 1 / the number of agents on the bike
+					utilityShare := 1 / totAgents
```

Comment:
I do think the utilityShare should simply be 1 / total agents and should be independent of the current energy levels. Then there's an incentive for agents to be mischevious and not spend energy/freeride etc. I was thinking the energy levels should be used in voting rather than for rewards